src/protean/core/field/association.py

```diff
@@ -0,0 +1,113 @@
+from .base import Field
+from .mixins import FieldCacheMixin
+
+from protean.core import exceptions
+
+
+class ReferenceField(Field):
+    """Shadow Attribute Field to back References"""
+
+    def __init__(self, reference, **kwargs):
+        """Accept reference field as a an attribute, otherwise is a straightforward field"""
+        self.reference = reference
+        super().__init__(**kwargs)
+
+    def __set__(self, instance, value):
+        """Override `__set__` to update relation field"""
+        value = self._load(value)
+
+        if value:
+            instance.__dict__[self.field_name] = value
+
+            # Fetch target object and refresh the reference field value
```

Comment:
We should not load this here and instead this needs to be lazy loaded. The reason I say this is that in cq project I have many cases where I am getting an entity for accessing concrete fields, now in this approach this will fire off queries equal to the number of reference fields which is not what i want. So it needs to be lazy